editor/editor_properties.cpp

```diff
@@ -2182,8 +2182,97 @@ void EditorPropertyResource::_menu_option(int p_which) {
+								inherits = "\"" + existing->get_path() + "\"";
+							}
+							break;
+						}
+					}
+				} else if (get_edited_property() == "script") {
+					// This is the script property of an object's sub-property, so inherit from that object's class
+					inherits = get_edited_object()->get_class();
+				} else {
+					inherits = "Reference";
+				}
+
+				String path;
+				// Path creation:
+				// Get the object's basename
+				//	If the object has no basename, get the scene root's filename
+				//		If the scene root has a filename, append the object's name/class to it
+				//		If it does not, append the object's name/class to "res://"
+				// If the property we're editing is not "script", append the property name to the filename
+				if (Object::cast_to<Node>(get_edited_object()) &&
+						EditorNode::get_singleton()->get_editor_data().get_edited_scene_root()->is_a_parent_of(Object::cast_to<Node>(get_edited_object()))) {
+					// This is a node in the scene tree, but it is not the script property
+					Node *edited_node = Object::cast_to<Node>(get_edited_object());
+
+					path = edited_node->get_filename().get_basename();
+					if (path == "") {
+						String root_path = EditorNode::get_singleton()->get_editor_data().get_edited_scene_root()->get_filename();
+						if (root_path == "") {
+							path = String("res://").plus_file(edited_node->get_name());
+						} else {
+							path = root_path.get_base_dir().plus_file(edited_node->get_name());
+						}
+					}
+				} else if (Object::cast_to<Resource>(get_edited_object())) {
+					RES edited_res = Object::cast_to<Resource>(get_edited_object());
+
+					path = edited_res->get_path().get_basename();
+					if (path == "") {
+						String root_path = EditorNode::get_singleton()->get_editor_data().get_edited_scene_root()->get_filename();
+						if (root_path == "") {
+							path = String("res://").plus_file(edited_res->get_name());
+						} else {
+							path = root_path.get_base_dir().plus_file(edited_res->get_name());
+						}
+					}
+				} else {
+					String root_path = EditorNode::get_singleton()->get_editor_data().get_edited_scene_root()->get_filename();
+					if (root_path == "") {
+						path = String("res://").plus_file(get_edited_object()->get_class());
+					} else {
+						path = root_path.get_base_dir().plus_file(get_edited_object()->get_class());
+					}
+				}
+
+				// If we're editing an object that isn't in the scene tree, we still have the script property,
+				// so we don't want to accidentally add the property name to the end of the script path.
+				if (get_edited_property() != "script") {
+					// path = <node_or_scene_name>.<property_name>.
+					// The extra period at the end is to prevent the property name from being seen as the type by ScriptCreateDialog
+					String path_property_name = get_edited_property();
+					path_property_name = path_property_name.replace(":", ".").replace("/", ".").replace("\\", ".");
+					path = path + "." + path_property_name + ".";
```

Comment:
It would be nice if it could get a path to the current property. For example, if the path to the property is `MyCSGMesh:material:albedo_texture:script`, the path could generated as `"res://scenes/MyCSGMesh.material.albedo_texture.gd"`. Preliminary testing seems to show that this would be a bit harder than I initially thought, however, and might be better served as its own PR anyways (since from what I can tell, I'd need to implement additional functionality into InspectorDock or something).